test/GraphingTest.hs

```diff
@@ -0,0 +1,20 @@
+module GraphingTest
+  ( spec_unfold
+  )
+  where
+
+import Test.Tasty.Hspec
+import Prelude
+import Graphing
+import GraphUtil
+
+spec_unfold :: Spec
+spec_unfold = do
+  describe "graphing unfold" $ do
+    it "should unfold deeply" $ do
+      let graph :: Graphing Int
+          graph = unfold [10] (\x -> if x > 0 then [x-2] else []) id
```

Comment:
the `id` here highlights that `unfold` has the wrong type. but it sure is convenient. (same issue as `withLabeling`)